### ISSUE TITLE
Cannot read properties of null (reading 'tagName')

### ISSUE THREAD

**Issue by AstRonin** [glennreyes/react-countup#736]
[x] Bug
[ ] Feature request

`"next": "13"`
`"countup": "2.3.2",`
`"react-countup": "^6.3.2",`
`"react": "18.2.0",`

## Description

Reproduce:

Open page with Count but do not start it.
Then move to other page (by **Next** routing).

@inorganik  in https://github.com/inorganik/countUp.js/issues/293 said me it is not bug of countup.

```
Unhandled Runtime Error
TypeError: Cannot read properties of null (reading 'tagName')

Call Stack
t.printValue
node_modules/countup.js/dist/countUp.min.js (1:4846)
t.reset
node_modules/countup.js/dist/countUp.min.js (1:4275)
eval
node_modules/react-countup/build/index.js (251:0)
eval
node_modules/react-countup/build/index.js (134:0)
eval
node_modules/react-countup/build/index.js (289:0)
safelyCallDestroy
node_modules/react-dom/cjs/react-dom.development.js (22932:0)
commitHookEffectListUnmount
node_modules/react-dom/cjs/react-dom.development.js (23100:0)
commitPassiveUnmountInsideDeletedTreeOnFiber
node_modules/react-dom/cjs/react-dom.development.js (25098:0)
commitPassiveUnmountEffectsInsideOfDeletedTree_begin
node_modules/react-dom/cjs/react-dom.development.js (25048:0)
commitPassiveUnmountEffects_begin
node_modules/react-dom/cjs/react-dom.development.js (24956:0)
commitPassiveUnmountEffects
node_modules/react-dom/cjs/react-dom.development.js (24941:0)
flushPassiveEffectsImpl
node_modules/react-dom/cjs/react-dom.development.js (27038:0)
flushPassiveEffects
node_modules/react-dom/cjs/react-dom.development.js (26984:0)
performSyncWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (26076:0)
flushSyncCallbacks
node_modules/react-dom/cjs/react-dom.development.js (12042:0)
commitRootImpl
node_modules/react-dom/cjs/react-dom.development.js (26959:0)
commitRoot
node_modules/react-dom/cjs/react-dom.development.js (26682:0)
finishConcurrentRender
node_modules/react-dom/cjs/react-dom.development.js (25981:0)
performConcurrentWorkOnRoot
node_modules/react-dom/cjs/react-dom.development.js (25809:0)
workLoop
node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js (266:0)
flushWork
node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js (239:0)
MessagePort.performWorkUntilDeadline
node_modules/react-dom/node_modules/scheduler/cjs/scheduler.development.js (533:0)
```

**Comment by scottcanoni** [comment]
I am also encountering this issue.  Is there a workaround?

**Comment by scottcanoni** [comment]
When I put a breakpoint on the error, I can see that the countup `el` is `null` and there is an error: "[CountUp] target is null or undefined"

![image](https://user-images.githubusercontent.com/2506805/202919560-389b3747-65fb-42ae-9c59-0a9f2a15f74d.png)

It seems that before you print the value, you need to make sure the element exists first.

**Comment by inorganik** [comment]
Fwiw this part of countUp.js has not changed for very long time. So perhaps this is a react change or router change. To fix, it seems to me that the countUp instance needs to be stopped or deleted when you navigate away. Since I didn't create the react version, I'm not sure the best way to do that. But my first thought is using a lifecycle hook like "component will unmount". Maybe @glennreyes can weigh in.

**Comment by glennreyes** [comment]
Not sure how to deal with that, maybe @mmarkelov as being the core maintainer has some insights here?

**Comment by mmarkelov** [comment]
hello @AstRonin @scottcanoni! Could you provide the small representation of this bug? I'lll try to look through it

**Comment by Echis65** [comment]
> I am also encountering this issue. Is there a workaround?

I think the workaround is not using the delay prop. I only had this error when i included a delay

**Comment by scottcanoni** [comment]
@mmarkelov I created this example which demonstrates the problem:
https://codesandbox.io/s/react-countup-react18-test-2pbz4n

While putting this together, I've been looking into it and I can confirm a few things:

1. The issue presents itself only when using `<React.StrictMode>`.  Removing that and it works fine.
2. Removing the delay also fixes it, like @Echis65 confirms.

**Comment by AstRonin** [comment]
Reproduced bug

- click to About or Contact Us

https://codesandbox.io/s/nextjs-page-transition-react-spring-forked-3f0n63?file=/pages/index.js

![image](https://user-images.githubusercontent.com/2378239/203725783-45ab9430-ca54-48fc-b3b7-0ecd6a311fc5.png)

**Comment by mmarkelov** [comment]
I just published **6.4.0** - this is just quick fix. I need more time to investigate it and provide better solution. Hope it will help

**Comment by scottcanoni** [comment]
> I just published **6.4.0** - this is just quick fix. I need more time to investigate it and provide better solution. Hope it will help

With version 6.4.0 there is a different error: 
Cannot read properties of null (reading 'tagName')
![image](https://user-images.githubusercontent.com/2506805/213614237-3acabcb7-ca3c-46da-bcf6-f98e5898d3c8.png)

Reproduced here:
https://codesandbox.io/s/competent-sound-e6dbn1?file=/src/_app.js

Though, for me, they still animate and count, just this JS error in the console.  I guess codesandbox.io is more strict.

**Comment by ahmadi-akbar** [comment]
same here

**Comment by scottcanoni** [comment]
@mmarkelov , I created this branch that replicates the issue.  I can now confirm it only happens when using `<React.StrictMode>` in combination with using the `start={0}` prop with any value.

https://github.com/glennreyes/react-countup/compare/master...scottcanoni:react-countup:bugfix-react18?expand=1

I think it may be due to the logic I changed but I am having trouble testing it in the demo setup.  Maybe you can assist?

This is suspect #1:

```
typeof props.start !== 'undefined' ? getCountUp().formattingFn(props.start) : ''
```

**Comment by scottcanoni** [comment]
My workaround right now is to simply remove the `start={0}` prop in my usage and it works fine without error and as expected.

**Comment by inorganik** [comment]
I can make a patch for CountUp which might fix it... I'll try to release before the end of the week.

**Comment by inorganik** [comment]
Just published 2.4.1. with a null check on the element... @mmarkelov can you update CountUp.js and see if that helps?

**Comment by mmarkelov** [comment]
@scottcanoni sorry, I don’t have much time right now, i'll look through it during the end of the week, hope that it will be fine with new CountUp version. I'll let you know. Thanks

**Comment by inorganik** [comment]
@mmarkelov I mean just run `npm update countup.js` and then publish a patch

**Comment by inorganik** [comment]
I created a gist for using CountUp.js directly with react in case anyone is blocked by this https://gist.github.com/inorganik/2cf776865a4c65c12857027870e9898e

**Comment by mmarkelov** [comment]
I just published the **reaact-counup@6.4.1** with actual **CountUp** version, so I guess it should help with this bug.
@inorganik thanks a lot for your help with this.
@scottcanoni could you please this new version and let me know if it works. I'll keep this issue opened for a while

**Comment by scottcanoni** [comment]
> @scottcanoni could you please this new version and let me know if it works. I'll keep this issue opened for a while

Confirmed fixed!  I tested with 6.4.1 and can confirm the error is gone here on CodeSandbox and in my personal project.

https://codesandbox.io/s/competent-sound-e6dbn1?file=/src/_app.js

Thanks and cheers!
